File: Data Structures & Algorithms/two-integer-sum/submission-4.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numIndexMap = new HashMap<>();

        // Iterate through every element to find target
        // Build numIndexMap along the way
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];

            if (numIndexMap.containsKey(diff)) {
                return new int[] {numIndexMap.get(diff), i};
            }

            numIndexMap.put(nums[i], i);
        }

        // return empty result to satisfy compiler
        // Since the problem says there is guaranteed to be exactly one solution,
        // so we will never return an empty array.
        return new int[] {};
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        m = defaultdict(int)
        result = []

        for n in nums:
            m[n] += 1

        m = sorted(m, key=lambda x: m[x])

        for i in range(len(m) - 1, -1, -1):
            if k == 0:
                return result
            result.append(m[i])
            k -= 1
        return result